xtask/src/release.rs

```diff
@@ -0,0 +1,186 @@
+use std::env;
+
+use clap::{Args, Subcommand, ValueEnum};
+use xshell::{cmd, pushd};
+
+use crate::{workspace, Result};
+
+#[derive(Args)]
+pub struct ReleaseArgs {
+    #[clap(subcommand)]
+    cmd: ReleaseCommand,
+}
+
+#[derive(PartialEq, Subcommand)]
+enum ReleaseCommand {
+    /// Prepare the release of the matrix-sdk workspace.
+    ///
+    /// This command will update the README, prepend the CHANGELOG.md file using
+    /// `git cliff`, and bump the versions in the `Cargo.toml` files.
+    Prepare {
+        /// What type of version bump we should perform.
+        version: ReleaseVersion,
+        /// Actually prepare a release. Dry-run mode is the default.
+        #[clap(long)]
+        execute: bool,
+    },
+    /// Publish the release.
+    ///
+    /// This command will create signed tags, publish the release on crates.io,
+    /// and finally push the tags to the repo.
+    Publish {
+        /// Actually publish a release. Dry-run mode is the default
```

Comment:
I would prefer `wet-run` rather than `execute` then 😆.

Reply:
It's the lingo `cargo-release` itself uses, for better or worse.